src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -395,10 +442,11 @@ impl StaticHeader {
             // Copy 1 read OK, 2 resulted in an IO error
             (Ok(buf_loc_1), Err(_)) => match StaticHeader::sigblock_from_buf(&buf_loc_1) {
                 Ok(loc_1) => {
-                    if loc_1.is_some() {
-                        BDA::write(f, &buf_loc_1, MetadataLocation::Second)?;
+                    if let Some(loc_1) = loc_1 {
```

Comment:
I think that the match arm here, and the one immediately below it can also be abstracted, kind of like suggestion above.

Reply:
Not as is. I'll change the function parameters to allow it.